agent/src/agent/pipeline/pipeline.py

```diff
@@ -132,47 +128,51 @@ def flush_bucket_size(self, value: str):
         self.config[self.FLUSH_BUCKET_SIZE] = FlushBucketSize(value).value
 
     @property
-    def static_dimension_names(self):
-        return self.static_dimensions.keys()
+    def static_dimensions(self) -> dict:
+        return self.config.get('properties', {})
+
+    @property
+    def static_dimension_names(self) -> list:
+        return [tools.replace_illegal_chars(s_dim) for s_dim in self.static_dimensions.keys()]
 
     @property
-    def dimensions(self) -> list:
-        dimensions = self.config.get('dimensions')
-        if not dimensions:
-            return []
-        if type(self.config['dimensions']) is dict:
-            dimensions = self.required_dimensions + self.optional_dimensions
-        return dimensions
+    def dimensions(self) -> list | dict:
+        return self.config.get('dimensions', [])
+
+    @property
+    def dimension_paths(self) -> list:
+        return [self._get_property_path(value) for value in self.all_dimensions]
 
     @property
     def required_dimensions(self) -> list:
-        dimensions = self.config.get('dimensions')
-        if type(dimensions) is not dict or 'required' not in dimensions:
+        if type(self.dimensions) is list:
             return []
-        return dimensions['required']
+        return self.dimensions.get('required', [])
 
     @property
-    def optional_dimensions(self) -> list:
-        dimensions = self.config.get('dimensions')
-        if type(dimensions) is not dict or 'optional' not in dimensions:
-            return []
-        return dimensions['optional']
+    def required_dimension_paths(self) -> list:
+        return [self._get_property_path(value) for value in self.required_dimensions]
 
     @property
-    def dimensions_names(self) -> list:
-        return [tools.replace_illegal_chars(d.replace('/', '_')) for d in self.dimensions]
+    def optional_dimensions(self) -> list:
+        if type(self.dimensions) is list:
+            return []
+        return self.dimensions.get('optional', [])
 
     @property
-    def dimensions_paths(self) -> list:
-        return [self._get_property_path(value) for value in self.dimensions]
+    def all_dimensions(self) -> list:
+        if not self.dimensions or type(self.dimensions) is list:
+            return self.dimensions + self.static_dimension_names
+        return self.required_dimensions + self.optional_dimensions + self.static_dimension_names
 
     @property
-    def required_dimensions_paths(self) -> list:
-        return [self._get_property_path(value) for value in self.required_dimensions]
+    def dimension_names(self) -> list:
+        # todo why replace / ? are you sure we need to replace it?
```

Comment:
when you have a nested json, user specifies a path to the property with `level1/level2/level3`. For some reason, I decided that we need to replace '/' with '_' so we can't change it because it will break metrics for old pipelines